packages/contracts/src/healthcheck/HealthChecker.sol

```diff
@@ -86,17 +100,76 @@ abstract contract HealthChecker is SafeInitializable, OwnableUpgradeable {
-            profit,
-            loss,
-            debtPayment,
-            debtOutstanding,
-            totalDebt
-        );
-        if (!success) {
+        // If no custom health check implementation provided, call default one.
+        uint8 checkResult = 0;
+        if (address(healthCheck) == address(0)) {
+            checkResult = _defaultCheck(
+                strategy,
+                profit,
+                loss,
+                totalDebt
+            );
+        } else {
+            // Perform the health check, to revert the transaction
+            checkResult = healthCheck.check(
+                strategy,
+                profit,
+                loss,
+                debtPayment,
+                debtOutstanding,
+                totalDebt,
+                gasCost
+            );
+        }
+
+        if(checkResult == 2) { // call fallback if loss to big
+            healthCheckFallback();
+        } else if(checkResult == 1) { // call secondary fallback if loss is acceptable
+            acceptableLossFallback();
+        }
+        // just allow to run normally in case of transaction is profitable
+    }
+
+    /// @dev Fallback function that is called when the health check fails.
+    function healthCheckFallback() internal virtual {
+        revert HealthCheckFailed();
+    }
+
+    /// @dev Fallback function that is called when the health check fails but loss is acceptable.
+    /* solhint-disable no-empty-blocks */
+    function acceptableLossFallback() internal virtual {
+        // do nothing by default but can be overridden
+    }
+    /* solhint-disable no-empty-blocks */
+
+    /// @notice Performs the health check by comparing execution strategy parameters. This behavior can be overridden with custom health check implementation.
+    /// @param profit The amount of funds that the strategy realised as profit.
+    /// @param loss The amount of funds that the strategy realised as loss.
+    /// @param totalDebt The total amount of funds borrowed by the strategy from the vault.
+    function _defaultCheck(
+        address strategy,
+        uint256 profit,
+        uint256 loss,
+        uint256 totalDebt
+    ) internal returns (uint8 result)
+    {
+        // If no target provided skipp the execution.
+        if (address(strategy) == address(0)) {
             revert HealthCheckFailed();
         }
+
+        if(profit > 0 || loss == 0) {
+            // if stategy profit is positive or zero, then it is healthy and we need to get this profit
+            result = 0;
+        } else if(loss > 0 && loss <= totalDebt * shutdownLossRatio / MAX_BPS) { //todo verify the condition
+            // if loss is positive but below critical loss threshold
+            result = 1;
+        } else {
+            // if loss is positive but still makes sense to run transaction to perform reporting.
+            result = 2;
+        }
+
+        emit HealthCheckTriggered(result);
+        return result;
```

Comment:
Please make a separate contract and move this logic into this HealthChecker contract. We want to modify the health check logic separately from strategies.

Reply:
I am really surprised to see this comment. The last time I showed you the code you were fine with the default health check logic that can be easily overridden with a custom health check from the method above.

Reply:
moved to external contract